common/src/fixpath.c

```diff
@@ -224,7 +246,7 @@ char *fix_at_file(char *in)
 
   tmpdir = getenv("TMP");
   if (tmpdir == NULL) {
-    tmpdir = "c:/cygwin/tmp";
+    tmpdir = ((stat("c:/cygwin64", &buf) == 0) && (buf.st_mode & S_IFDIR != 0)) ? "c:/cygwin64/tmp" : "c:/cygwin/tmp";
```

Comment:
Can you move the `struct stat buf;` line to this block?  Variables should have the shortest possible life times to make it easier to reason about the code.

Reply:
Something like says: try and use `c:/cygwin/tmp` if it exists, otherwise assume cygwin64:

```c
if (tmpdir == NULL) {
  tmpdir = "c:/cygwin/tmp";
  if ((stat("c:/cygwin", &buf) != 0) || (buf.st_mode & S_IFDIR == 0)) {
    tmpdir = "c:/cygwin64/tmp";
  }
}
```

Reply:
fixed as suggested

Reply:
@DanHeidinga - ready for review